kv/kvserver: TestLeaseExpirationBelowFutureTimeRequest failed

kv/kvserver.TestLeaseExpirationBelowFutureTimeRequest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21112736?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21112736?buildTab=artifacts#/) on master @ [c2323e2129492ed8f065b5d498df9422e7b97e89](https://github.com/cockroachdb/cockroach/commits/c2323e2129492ed8f065b5d498df9422e7b97e89):


```
=== RUN   TestLeaseExpirationBelowFutureTimeRequest
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestLeaseExpirationBelowFutureTimeRequest663648490
    test_log_scope.go:82: use -show-logs to present logs inline
    client_replica_test.go:2045: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestLeaseExpirationBelowFutureTimeRequest663648490
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest (1.80s)
=== RUN   TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
    test_server_shim.go:679: DRPC is enabled (via override by TestingGlobalDRPCOption)
    client_replica_test.go:2037: 
        	Error Trace:	pkg/kv/kvserver/client_replica_test.go:2037
        	            				pkg/testutils/subtest.go:23
        	Error:      	Expected nil, but got: &kvpb.Error{UnexposedTxn:(*roachpb.Transaction)(nil), OriginNode:0, EncodedError:errorspb.EncodedError{Error:(*errorspb.EncodedError_Leaf)(0x4002c47730)}, Index:(*kvpb.ErrPosition)(nil), Now:hlc.ClockTimestamp{WallTime:1771337743566954651, Logical:9}}
        	Test:       	TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false (0.84s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseExpirationBelowFutureTimeRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60463

kv/kvserver.TestLeaseExpirationBelowFutureTimeRequest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21164582?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21164582?buildTab=artifacts#/) on master @ [c921910f92fa774a2b88ad2c0a22e74c4c700a4b](https://github.com/cockroachdb/cockroach/commits/c921910f92fa774a2b88ad2c0a22e74c4c700a4b):


```
=== RUN   TestLeaseExpirationBelowFutureTimeRequest
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestLeaseExpirationBelowFutureTimeRequest1519903610
    test_log_scope.go:82: use -show-logs to present logs inline
    client_replica_test.go:2045: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestLeaseExpirationBelowFutureTimeRequest1519903610
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest (1.56s)
=== RUN   TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
    test_server_shim.go:686: DRPC is enabled (via override by TestingGlobalDRPCOption)
    client_replica_test.go:2037: 
        	Error Trace:	pkg/kv/kvserver/client_replica_test.go:2037
        	            				pkg/testutils/subtest.go:23
        	Error:      	Expected nil, but got: &kvpb.Error{UnexposedTxn:(*roachpb.Transaction)(nil), OriginNode:0, EncodedError:errorspb.EncodedError{Error:(*errorspb.EncodedError_Leaf)(0x40112313a8)}, Index:(*kvpb.ErrPosition)(nil), Now:hlc.ClockTimestamp{WallTime:1772131062128163136, Logical:9}}
        	Test:       	TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false (0.73s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseExpirationBelowFutureTimeRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseExpirationBelowFutureTimeRequest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21184419?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21184419?buildTab=artifacts#/) on master @ [3a95e6305dace4953943bd2542941068d9a48633](https://github.com/cockroachdb/cockroach/commits/3a95e6305dace4953943bd2542941068d9a48633):


```
=== RUN   TestLeaseExpirationBelowFutureTimeRequest
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestLeaseExpirationBelowFutureTimeRequest4108164818
    test_log_scope.go:82: use -show-logs to present logs inline
    client_replica_test.go:2045: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestLeaseExpirationBelowFutureTimeRequest4108164818
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest (1.93s)
=== RUN   TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
    client_replica_test.go:2037: 
        	Error Trace:	pkg/kv/kvserver/client_replica_test.go:2037
        	            				pkg/testutils/subtest.go:23
        	Error:      	Expected nil, but got: &kvpb.Error{UnexposedTxn:(*roachpb.Transaction)(nil), OriginNode:0, EncodedError:errorspb.EncodedError{Error:(*errorspb.EncodedError_Leaf)(0x4009284ca8)}, Index:(*kvpb.ErrPosition)(nil), Now:hlc.ClockTimestamp{WallTime:1772502153017082830, Logical:9}}
        	Test:       	TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false (0.87s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseExpirationBelowFutureTimeRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseExpirationBelowFutureTimeRequest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21250960?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21250960?buildTab=artifacts#/) on master @ [86c176dbb4e93eda87764ced207d046e81057e14](https://github.com/cockroachdb/cockroach/commits/86c176dbb4e93eda87764ced207d046e81057e14):


```
=== RUN   TestLeaseExpirationBelowFutureTimeRequest
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestLeaseExpirationBelowFutureTimeRequest2236457571
    test_log_scope.go:82: use -show-logs to present logs inline
    client_replica_test.go:2045: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestLeaseExpirationBelowFutureTimeRequest2236457571
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest (2.20s)
=== RUN   TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:638: DRPC is enabled (via testserver factory defaults)
    client_replica_test.go:2037: 
        	Error Trace:	pkg/kv/kvserver/client_replica_test.go:2037
        	            				pkg/testutils/subtest.go:23
        	Error:      	Expected nil, but got: &kvpb.Error{UnexposedTxn:(*roachpb.Transaction)(nil), OriginNode:0, EncodedError:errorspb.EncodedError{Error:(*errorspb.EncodedError_Leaf)(0x400efa0700)}, Index:(*kvpb.ErrPosition)(nil), Now:hlc.ClockTimestamp{WallTime:1773441540197707475, Logical:9}}
        	Test:       	TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false (1.09s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseExpirationBelowFutureTimeRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseExpirationBelowFutureTimeRequest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21252003?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21252003?buildTab=artifacts#/) on master @ [3451fd7020697c89b1bf726a8eb49ef9918996ad](https://github.com/cockroachdb/cockroach/commits/3451fd7020697c89b1bf726a8eb49ef9918996ad):


```
=== RUN   TestLeaseExpirationBelowFutureTimeRequest
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestLeaseExpirationBelowFutureTimeRequest2218898285
    test_log_scope.go:82: use -show-logs to present logs inline
    client_replica_test.go:2045: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestLeaseExpirationBelowFutureTimeRequest2218898285
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest (1.62s)
=== RUN   TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:638: DRPC is enabled (via testserver factory defaults)
    client_replica_test.go:2037: 
        	Error Trace:	pkg/kv/kvserver/client_replica_test.go:2037
        	            				pkg/testutils/subtest.go:23
        	Error:      	Expected nil, but got: &kvpb.Error{UnexposedTxn:(*roachpb.Transaction)(nil), OriginNode:0, EncodedError:errorspb.EncodedError{Error:(*errorspb.EncodedError_Leaf)(0x401679c338)}, Index:(*kvpb.ErrPosition)(nil), Now:hlc.ClockTimestamp{WallTime:1773459035624399599, Logical:9}}
        	Test:       	TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false (0.78s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseExpirationBelowFutureTimeRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseExpirationBelowFutureTimeRequest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21265405?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21265405?buildTab=artifacts#/) on master @ [6e210ba6aa33cea5e27b1a8fae212c27941781f4](https://github.com/cockroachdb/cockroach/commits/6e210ba6aa33cea5e27b1a8fae212c27941781f4):


```
=== RUN   TestLeaseExpirationBelowFutureTimeRequest
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestLeaseExpirationBelowFutureTimeRequest2739348479
    test_log_scope.go:82: use -show-logs to present logs inline
    client_replica_test.go:2045: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestLeaseExpirationBelowFutureTimeRequest2739348479
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest (2.06s)
=== RUN   TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    test_server_shim.go:638: DRPC is enabled (via testserver factory defaults)
    client_replica_test.go:2037: 
        	Error Trace:	pkg/kv/kvserver/client_replica_test.go:2037
        	            				pkg/testutils/subtest.go:23
        	Error:      	Expected nil, but got: &kvpb.Error{UnexposedTxn:(*roachpb.Transaction)(nil), OriginNode:0, EncodedError:errorspb.EncodedError{Error:(*errorspb.EncodedError_Leaf)(0x40138be7e0)}, Index:(*kvpb.ErrPosition)(nil), Now:hlc.ClockTimestamp{WallTime:1773772655114452566, Logical:9}}
        	Test:       	TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false (1.04s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseExpirationBelowFutureTimeRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseExpirationBelowFutureTimeRequest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21328922?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21328922?buildTab=artifacts#/) on master @ [7f06461ba3bff8b69959abb163918b9afc5a3fd5](https://github.com/cockroachdb/cockroach/commits/7f06461ba3bff8b69959abb163918b9afc5a3fd5):


```
=== RUN   TestLeaseExpirationBelowFutureTimeRequest
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/15bcc1d4cfe2c01885bd956ef4f14c71/logTestLeaseExpirationBelowFutureTimeRequest840120545
    test_log_scope.go:82: use -show-logs to present logs inline
    client_replica_test.go:2045: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/15bcc1d4cfe2c01885bd956ef4f14c71/logTestLeaseExpirationBelowFutureTimeRequest840120545
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest (1.53s)
=== RUN   TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    client_replica_test.go:2037: 
        	Error Trace:	pkg/kv/kvserver/client_replica_test.go:2037
        	            				pkg/testutils/subtest.go:23
        	Error:      	Expected nil, but got: &kvpb.Error{UnexposedTxn:(*roachpb.Transaction)(nil), OriginNode:0, EncodedError:errorspb.EncodedError{Error:(*errorspb.EncodedError_Leaf)(0x1f2a03b86958)}, Index:(*kvpb.ErrPosition)(nil), Now:hlc.ClockTimestamp{WallTime:1777331379902455972, Logical:9}}
        	Test:       	TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false (0.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseExpirationBelowFutureTimeRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseExpirationBelowFutureTimeRequest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21329019?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21329019?buildTab=artifacts#/) on master @ [3399ae7c9d6e7b7c840a2355f9fbc50a4cf28ffe](https://github.com/cockroachdb/cockroach/commits/3399ae7c9d6e7b7c840a2355f9fbc50a4cf28ffe):


```
=== RUN   TestLeaseExpirationBelowFutureTimeRequest
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/15bcc1d4cfe2c01885bd956ef4f14c71/logTestLeaseExpirationBelowFutureTimeRequest3568815378
    test_log_scope.go:82: use -show-logs to present logs inline
    client_replica_test.go:2045: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/15bcc1d4cfe2c01885bd956ef4f14c71/logTestLeaseExpirationBelowFutureTimeRequest3568815378
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest (1.55s)
=== RUN   TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    client_replica_test.go:2037: 
        	Error Trace:	pkg/kv/kvserver/client_replica_test.go:2037
        	            				pkg/testutils/subtest.go:23
        	Error:      	Expected nil, but got: &kvpb.Error{UnexposedTxn:(*roachpb.Transaction)(nil), OriginNode:0, EncodedError:errorspb.EncodedError{Error:(*errorspb.EncodedError_Leaf)(0x48d9751058f8)}, Index:(*kvpb.ErrPosition)(nil), Now:hlc.ClockTimestamp{WallTime:1777335818737249992, Logical:9}}
        	Test:       	TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false (0.76s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseExpirationBelowFutureTimeRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseExpirationBelowFutureTimeRequest [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21330354?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21330354?buildTab=artifacts#/) on master @ [3fe3c2cc428b2674511a41ff3ebd2f31b8a10b80](https://github.com/cockroachdb/cockroach/commits/3fe3c2cc428b2674511a41ff3ebd2f31b8a10b80):


```
=== RUN   TestLeaseExpirationBelowFutureTimeRequest
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/15bcc1d4cfe2c01885bd956ef4f14c71/logTestLeaseExpirationBelowFutureTimeRequest3353135058
    test_log_scope.go:82: use -show-logs to present logs inline
    client_replica_test.go:2045: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/15bcc1d4cfe2c01885bd956ef4f14c71/logTestLeaseExpirationBelowFutureTimeRequest3353135058
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest (3.01s)
=== RUN   TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    client_replica_test.go:2037: 
        	Error Trace:	pkg/kv/kvserver/client_replica_test.go:2037
        	            				pkg/testutils/subtest.go:23
        	Error:      	Expected nil, but got: &kvpb.Error{UnexposedTxn:(*roachpb.Transaction)(nil), OriginNode:0, EncodedError:errorspb.EncodedError{Error:(*errorspb.EncodedError_Leaf)(0x5187de3345f8)}, Index:(*kvpb.ErrPosition)(nil), Now:hlc.ClockTimestamp{WallTime:1777404436891637376, Logical:9}}
        	Test:       	TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false (0.92s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseExpirationBelowFutureTimeRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseExpirationBelowFutureTimeRequest [failed](https://mesolite.cluster.engflow.com/invocations/default/a3c36f83-1ddb-4702-a9eb-73fa82f4c239?testReportRun=2&testReportShard=5&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [59267e5d249f07e210c2a3b2b66c5655006d0e0e](https://github.com/cockroachdb/cockroach/commits/59267e5d249f07e210c2a3b2b66c5655006d0e0e):


```
=== RUN   TestLeaseExpirationBelowFutureTimeRequest
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestLeaseExpirationBelowFutureTimeRequest3621825651
    test_log_scope.go:82: use -show-logs to present logs inline
    client_replica_test.go:2045: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseExpirationBelowFutureTimeRequest3621825651
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest (3.31s)
=== RUN   TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
    test_server_shim.go:146: using test tenant configuration from testserver factory defaults
    client_replica_test.go:2037: 
        	Error Trace:	pkg/kv/kvserver/client_replica_test.go:2037
        	            				pkg/testutils/subtest.go:23
        	Error:      	Expected nil, but got: &kvpb.Error{UnexposedTxn:(*roachpb.Transaction)(nil), OriginNode:0, EncodedError:errorspb.EncodedError{Error:(*errorspb.EncodedError_Leaf)(0x10721e043720)}, Index:(*kvpb.ErrPosition)(nil), Now:hlc.ClockTimestamp{WallTime:1777449609681201416, Logical:9}}
        	Test:       	TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false
--- FAIL: TestLeaseExpirationBelowFutureTimeRequest/tooFarInFuture=false (1.66s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=5</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseExpirationBelowFutureTimeRequest.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/client_replica_test.go
+++ b/pkg/kv/kvserver/client_replica_test.go
@@ -2000,6 +2000,25 @@ func TestLeaseExpirationBelowFutureTimeRequest(t *testing.T) {
 			return lease.Replica.StoreID == l.replica1.StoreID()
 		}, 5*time.Second, 100*time.Millisecond, "timed out waiting for replica 0 to pick up new lease")
 
+		// Wait for raft leadership to transfer to the leaseholder. The lease
+		// extension below goes through verifyLeaseRequestSafetyRLocked which
+		// treats a stasis-period lease as expired (PrevLeaseExpired=true) and
+		// falls through to verifyAcquisition, which rejects if we're not the
+		// raft leader. Without this wait, the raft group can be leaderless
+		// after AdminTransferLease initiates a leadership transfer.
+		require.Eventually(t, func() bool {
+			raftStatus := l.replica1.RaftStatus()
+			if raftStatus == nil {
+				return false
+			}
+			if raftStatus.RaftState == raftpb.StateLeader {
+				return true
+			}
+			t.Logf("replica1 raft state: %s, lead: %d, term: %d",
+				raftStatus.RaftState, raftStatus.Lead, raftStatus.Term)
+			return false
+		}, 5*time.Second, 100*time.Millisecond, "timed out waiting for raft leadership transfer to replica1")
+
 		// Pause the cluster's clocks.
 		l.manualClock.Pause()
 		atPause := l.manualClock.UnixNano()
@@ -2033,8 +2052,8 @@ func TestLeaseExpirationBelowFutureTimeRequest(t *testing.T) {
 			// may end up being extended by some other request. That fact that
 			// our request was rejected is good enough.
 		} else {
-			// The request should have been rejected.
-			require.Nil(t, pErr)
+			// The request should succeed and trigger a lease extension.
+			require.Nil(t, pErr, "expected request to succeed: %s", pErr)
 
 			// The lease should have been extended.
 			l.checkHasLease(t, 1)

--- a/pkg/kv/kvserver/client_replica_test.go
+++ b/pkg/kv/kvserver/client_replica_test.go
@@ -2006,18 +2006,17 @@ func TestLeaseExpirationBelowFutureTimeRequest(t *testing.T) {
 		// falls through to verifyAcquisition, which rejects if we're not the
 		// raft leader. Without this wait, the raft group can be leaderless
 		// after AdminTransferLease initiates a leadership transfer.
-		require.Eventually(t, func() bool {
+		testutils.SucceedsSoon(t, func() error {
 			raftStatus := l.replica1.RaftStatus()
 			if raftStatus == nil {
-				return false
+				return errors.New("raft status not available")
 			}
 			if raftStatus.RaftState == raftpb.StateLeader {
-				return true
+				return nil
 			}
-			t.Logf("replica1 raft state: %s, lead: %d, term: %d",
+			return errors.Newf("replica1 raft state: %s, lead: %d, term: %d",
 				raftStatus.RaftState, raftStatus.Lead, raftStatus.Term)
-			return false
-		}, 5*time.Second, 100*time.Millisecond, "timed out waiting for raft leadership transfer to replica1")
+		})
 
 		// Pause the cluster's clocks.
 		l.manualClock.Pause()

